distributed/scheduler.py

```diff
@@ -4644,10 +4653,99 @@ def update_graph_hlg(
         fifo_timeout=0,
         code=None,
     ):
-        unpacked_graph = HighLevelGraph.__dask_distributed_unpack__(hlg)
-        dsk = unpacked_graph["dsk"]
-        dependencies = unpacked_graph["deps"]
-        annotations = unpacked_graph["annotations"]
+        try:
+            graph: HighLevelGraph = pickle.loads(graph_header, buffers=graph_frames)
+        except Exception as e:
+            text = str(e)
+            exc = pickle.dumps(e)
+            parent: SchedulerState = cast(SchedulerState, self)
+            for key in keys:
+                ts = parent.new_task(
+                    key, None, "erred", computation=None  # computation
+                )
+                ts._exception = exc
+                ts._exception_text = text
+                ts._exception_blame = ts
+                parent.tasks[key] = ts
+
+            self.client_desires_keys(keys=keys, client=client)
+            return
+
+        dsk = dict(graph)
```

Comment:
Ideally low level graph optimization mostly goes away I think.  But in principle I agree that it can be done here.  We'll want to be mindful of annotations, but we would need to do that anyway.

I'm hopeful though that high level graph optimization removes the need here.